File: 백준/Bronze/1157. 단어 공부/단어 공부.py
word = input().upper()
word_set = list(set(word))
word_count = []

for i in word_set:
    cnt = word.count(i)
    word_count.append(cnt)

if word_count.count(max(word_count)) >= 2:
    print('?')
else:
    print(word_set[word_count.index(max(word_count))])

File: 백준/Bronze/1157. 단어 공부/단어 공부.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];
rl.on("line", (line) => {
  input.push(line.trim());
  rl.close();
}).on("close", () => {
  const word = input[0].toString().toUpperCase();
  const count = Array(26).fill(0);

  for (let char of word) {
    count[char.charCodeAt() - 65] += 1;
  }

  const max = Math.max(...count);
  const maxCount = count.filter((v) => v === max).length;

  console.log(
    maxCount >= 2 ? "?" : String.fromCharCode(count.indexOf(max) + 65)
  );
  process.exit();
});
